src/sentry/static/sentry/app/views/settings/components/settingsBreadcrumb/projectCrumb.jsx

```diff
@@ -24,13 +26,27 @@ class ProjectCrumb extends React.Component {
     route: PropTypes.object,
   };
 
+  handleSelect = item => {
+    let {routes, params} = this.props;
+
+    let lastRoute = routes[routes.length - 1];
+    // We have to make exceptions for routes like "Project Alerts Rule Edit" or "Client Key Details"
+    // Since these models are project specific, we need to traverse up a route when switching projects
+    let stepBack = ROUTE_PATH_EXCEPTIONS.has(lastRoute.path) ? -1 : undefined;
```

Comment:
What does a stepBack value of `undefined` mean - would it not be `-2` in that case i.e. 2 steps back?

Reply:
`undefined` means it returns the current path.

We're going from something like:

```
[
  {path: '/', childRoutes: []},
  {path: '/settings/', name: 'Settings', indexRoute: {}, childRoutes: []},
  {name: 'Organizations', path: ':orgId/', childRoutes: []},
  {name: 'Projects', path: ':projectId/', childRoutes: []},
  {name: 'Rules', path: 'rules/'},
  {name: 'Rule Details', path: ':ruleId/'},
]
```

So if we're on the details page, we want to end up on `Rules` page (-1)